src/explore-education-statistics-common/src/components/form/__tests__/FormComboBox.test.tsx

```diff
@@ -1,804 +1,812 @@
 import React from 'react';
-import { fireEvent, render } from '@testing-library/react';
+import { fireEvent, render, screen, waitFor } from '@testing-library/react';
+import noop from 'lodash/noop';
+import renderer from 'react-test-renderer';
+import userEvent from '@testing-library/user-event';
 import FormComboBox from '../FormComboBox';
 
 describe('FormComboBox', () => {
   test('renders with no options by default', () => {
-    const { container } = render(
+    render(
       <FormComboBox
         id="test-combobox"
         inputLabel="Choose option"
-        onInputChange={() => {}}
-        onSelect={() => {}}
+        onInputChange={noop}
+        onSelect={noop}
         options={['Option 1', 'Option 2', 'Option 3']}
       />,
     );
 
-    const options = container.querySelectorAll('[role="option"]');
+    const options = screen.queryAllByRole('option');
 
     expect(options).toHaveLength(0);
-    expect(container.innerHTML).toMatchSnapshot();
+
+    const tree = renderer
+      .create(
+        <FormComboBox
+          id="test-combobox"
+          inputLabel="Choose option"
+          onInputChange={noop}
+          onSelect={noop}
+          options={['Option 1', 'Option 2', 'Option 3']}
+        />,
+      )
+      .toJSON();
+
+    expect(tree).toMatchSnapshot();
   });
 
-  test('renders with options in correct order when input changes', () => {
-    const { container, getByLabelText } = render(
+  test('renders with options in correct order when input changes', async () => {
+    render(
       <FormComboBox
         id="test-combobox"
         inputLabel="Choose option"
-        onInputChange={() => {}}
-        onSelect={() => {}}
+        onInputChange={noop}
+        onSelect={noop}
         options={['Option 1', 'Option 2', 'Option 3']}
       />,
     );
 
-    const input = getByLabelText('Choose option') as HTMLInputElement;
+    await userEvent.type(screen.getByLabelText('Choose option'), 'Test');
 
-    fireEvent.change(input, {
-      target: {
-        value: 'Test',
-      },
-    });
-
-    const options = container.querySelectorAll('[role="option"]');
+    const options = screen.queryAllByRole('option');
 
     expect(options).toHaveLength(3);
     expect(options[0]).toHaveTextContent('Option 1');
     expect(options[1]).toHaveTextContent('Option 2');
     expect(options[2]).toHaveTextContent('Option 3');
 
-    expect(container.innerHTML).toMatchSnapshot();
+    const tree = renderer
+      .create(
+        <FormComboBox
+          id="test-combobox"
+          inputLabel="Choose option"
+          onInputChange={noop}
+          onSelect={noop}
+          options={['Option 1', 'Option 2', 'Option 3']}
+        />,
+      )
+      .toJSON();
+
+    expect(tree).toMatchSnapshot();
```

Comment:
Same question as earlier - not sure why we're doing this? We can just use `container` from the earlier `render` to check the snapshot HTML

Reply:
Cool, didn't realize that we could use both the destructuring way of doing things as well as using `screen` from `@testing-library`